app/src/main/java/homer/impl/outlets/Outlet.java

```diff
@@ -0,0 +1,115 @@
+package homer.impl.outlets;
+
+import java.util.Objects;
+
+import homer.api.AdjustableDevice;
+import homer.api.DeviceInfo;
+
+/**
+ * Models electrical outlets of the house.
+ * 
+ * @param <S> Generic type for plugged device, an Outlet doesn't need a device to be plugged in,
+ * since it can be used with devices outside the emulated environment.
+ * 
+ * @author Alessandro Monticelli
+ */
+
+public class Outlet<S> implements AdjustableDevice<Double> {
+
+    private final DeviceInfo info;
+    private double state;
+    private final double minValue;
+    private final double maxValue;
+    private S device;
```

Comment:
The device should probably have type Device<?>, here you are saying the outlet can have a device of type S which can be anything.
Remember to make it an optional in the future.

Reply:
i agree with simone